### PR TITLE
update to react-router 6

### PR DESCRIPTION
**Pull request by Fullstop000**
UCP https://github.com/pingcap-incubator/tidb-dashboard/issues/249

Update to react-router6

Signed-off-by: Fullstop000 <fullstop1005@gmail.com>

### CODE REVIEW

**Comment by baurine** [comment]
hi @Fullstop000 , thanks for your contribution, I will test and review soon.

**Comment by baurine** [comment]
Hi @Fullstop000 , after my test,  it have some errors, statements / diagnose / search logs / profile pages can't show normally, they are all empty, and the console output following errors:

```
Uncaught Error: Route(...): Nothing was returned from render. This usually means a return statement is missing. Or, to render nothing, return null.
```

image

File: ui/packages/statement/src/RootComponent.tsx [hunk 1]
```diff
@@ -39,7 +40,7 @@ const App = withRouter((props) => {
           </Breadcrumb>
         </div>
         <div style={{ margin: 12 }}>
-          <Switch>
+          <Routes>
             <Route path="/statement/overview">
               <StatementsOverviewPage
                 dashboardClient={client.getInstance()}
```
Comment (baurine) [inline ui/packages/statement/src/RootComponent.tsx]: After my test, nest the component inside the `<Route>` doesn't work in v6, we need to make the component as `element` prop value, like this: `<Route path="/statement/overview" element={<StatementsOverviewPage .../>}`

**Comment by Fullstop000** [comment]
Got something pained after upgrading to react-router v6
image

**Comment by baurine** [comment]
> Got something pained after upgrading to react-router v6
> image

yep, that's a problem, we need to have a look.

**Comment by Fullstop000** [comment]
And I got this warning only when unmount `AppStatement`
image

**Comment by baurine** [comment]
> And I got this warning only when unmount `AppStatement`
> image

I will have a look today.

**Comment by baurine** [comment]
There is some discussion in https://github.com/facebook/react/issues/18178 , I tried to upgrade the  react to 16.13.1 , it doesn't dismiss the warning.

It should be caused by the new version react-router, above warning doesn't happen for react-router v5, someone has already reported in https://github.com/ReactTraining/react-router/issues/7199 as well.

So I think let's wait a few days more to see whether react-router will fix it, if it doesn't, we can merge it first and upgrade it in the future.

**Comment by Fullstop000** [comment]
> So I think let's wait a few days more to see whether react-router will fix it, if it doesn't, we can merge it first and upgrade it in the future.

Sounds reasonable :)

**Comment by baurine** [comment]
hi  @Fullstop000 , react-router fixed above warning in https://github.com/ReactTraining/react-router/releases/tag/v6.0.0-alpha.3 , you can have a try, but unfortunately they removed `<Redirect/>` component as well which we used, not sure whether it is easy to migrate, if it doesn't, I suggest let update react-router v6 until it is stable, @breeswish , how do you think about it?

**Comment by Fullstop000** [comment]
Maybe using a history API instead `<Redirect/>` ? I'll have a try later today.

**Comment by breeswish** [comment]
@baurine I'm fine with both. This task is not ergant.

**Comment by Fullstop000** [comment]
@baurine The second warning has been fixed and the first warning only shows on the initial render.

**Pull request merged**